tests/handlers/test_message.py

```diff
@@ -156,6 +158,79 @@ def test_duplicated_txn_id_one_call(self):
         self.assertEqual(len(events), 2)
         self.assertEqual(events[0].event_id, events[1].event_id)
 
+    def test_create_empty_prev_events_in_msc2716_room_version(self):
+        """Try to create an event without any prev_events (only auth_events).
+
+        This is currently only supported in the experimental MSC2716 room versions.
+        """
+        room_id = self.helper.create_room_as(
+            self.user_id,
+            tok=self.access_token,
+            room_version=RoomVersions.MSC2716v4.identifier,
+        )
+
+        # Create a member event we can use as an auth_event
+        memberEvent, memberEventContext = self.get_success(
+            create_event(
+                self.hs,
+                room_id=room_id,
+                type="m.room.member",
+                sender=self.requester.user.to_string(),
+                state_key=self.requester.user.to_string(),
+                content={"membership": "join"},
+            )
+        )
+        self.get_success(
+            self.persist_event_storage.persist_event(memberEvent, memberEventContext)
+        )
+
+        # Try to create the event with empty prev_events (only auth_events)
+        event, _ = self.get_success(
+            self.handler.create_event(
+                self.requester,
+                {
+                    "type": EventTypes.Message,
+                    "room_id": room_id,
+                    "sender": self.requester.user.to_string(),
+                    "content": {"msgtype": "m.text", "body": random_string(5)},
+                },
+                # Empty prev_events is the key thing we're testing here
+                prev_event_ids=[],
+                auth_event_ids=[memberEvent.event_id],
+            )
+        )
+        self.assertIsNotNone(event)
+
+    def test_reject_empty_prev_events_and_auth_events_in_msc2716_room_version(
+        self,
+    ):
+        """Try to create an event without any prev_events (only auth_events).
+
+        This is currently only supported in the experimental MSC2716 room versions.
+        """
+        room_id = self.helper.create_room_as(
+            self.user_id,
+            tok=self.access_token,
+            room_version=RoomVersions.MSC2716v4.identifier,
+        )
+
+        # Try to create the event with empty prev_events and empty auth_events
+        self.get_failure(
+            self.handler.create_event(
+                self.requester,
+                {
+                    "type": EventTypes.Message,
+                    "room_id": room_id,
+                    "sender": self.requester.user.to_string(),
+                    "content": {"msgtype": "m.text", "body": random_string(5)},
+                },
+                prev_event_ids=[],
+                # The event should be rejected when there are no auth_events
+                auth_event_ids=[],
+            ),
+            AssertionError,
+        )
+
```

Comment:
> It would be nice to have some tests for this, unless it's expected to be covered by https://github.com/matrix-org/complement/pull/214?

All of the MSC2716 Complement tests cover this use case since we use it so much there. I've just added some integration tests for this.

I didn't add tests for every case to be completely exhaustive but did make sure this extra functionality works and that we don't allow events with empty `auth_events`

 - Creating a `m.room.create` event
    - :x: In MSC2716 room version
    - :x: In existing stable room version
 - Creating event with empty `prev_event`
    - ✅ In MSC2716 room version
    - :x: In existing stable room version
 - Creating event with empty `prev_event` and empty `auth_event`  (make sure rejected)
    - ✅ In MSC2716 room version
    - :x: In existing stable room version
 - Creating event with `prev_events`
    - :x: In MSC2716 room version
    - :x: In existing stable room version
 - Creating event with `prev_events` but empty `auth_event` (make sure rejected)
    - :x: In MSC2716 room version
    - :x: In existing stable room version